bluehalo/node-rest-starter
# Feat/feedback update

This PR contains modifications to the `Feedback` routes, controller, service, and model. This is being done with the intention of expanding the `Feedback` functionality to be more like other admin services (like user service) and add some new functionality as well.  There is an accompanying PR for the `ngx-starter` with the UI changes that would use this server functionality (https://github.com/Asymmetrik/ngx-starter/pull/192).

**Change Summary:**
- Added 3 new fields to `Feedback` model, `status`, `assignee`, and `updated`. These are made with the intention of allowing a sort of in-app tracking system to allow admins to organize handling feedback.
- Added 2 new routes and accompanying controller and service methods to update `status` and `feedback` individually.
- Modified CSV export to use a column config posted with the export request instead of defaulting to including all columns. This allows users to select which columns they want exported based on what columns they have showing in the UI (modeled after `User` service).
- Modified the search to take in filter queries to enable UI Quick Filters (modeled after `User` service).

**Verification Steps**
While these updates can be verified using Postman or something, it's easier to do it through the UI. See https://github.com/Asymmetrik/ngx-starter/pull/192 for verification steps that include this functionality.

Let me know what I can improve! Thanks!
